lib/cog/command/pipeline/executor.ex

```diff
@@ -231,20 +231,11 @@ defmodule Cog.Command.Pipeline.Executor do
         resp = Cog.Command.Response.decode!(payload)
         case resp.status do
           "ok" ->
-            collected_output = case resp.body do
-                                 nil ->
-                                   # If there wasn't any output,
-                                   # there's nothing to collect
-                                   state.output
-                                 body ->
-                                   state.output ++ Enum.map(List.wrap(body),
-                                                            &store_with_template(&1, resp.template))
-                                   # body may be a map or a list
-                                   # of maps; if the latter, we
-                                   # want to accumulate it into
-                                   # one flat list
-                               end
+            collected_output = collect_output(resp, state.output)
             {:next_state, :execute_plan, %{state | output: collected_output}, 0}
+          "abort" ->
+            collected_output = collect_output(resp, state.output)
```

Comment:
I believe this collected output can theoretically include successful responses as well as the failing one (e.g., the first few iterations were good, but the last one failed).

Is this intentional, or do we only want to template a response for just the failing case?